sdk/src/manifest.rs

```diff
@@ -536,9 +536,12 @@ impl Manifest {
             self.set_title(ingredient.title());
         }
 
-        #[cfg(feature = "add_thumbnails")]
-        if let Ok((format, image)) = crate::utils::thumbnail::make_thumbnail(path.as_ref()) {
-            self.set_thumbnail(format, image);
+        // if a thumbnail is not already defined, create one here
+        if self.thumbnail().is_none() {
+            #[cfg(feature = "add_thumbnails")]
+            if let Ok((format, image)) = crate::utils::thumbnail::make_thumbnail(path.as_ref()) {
```

Comment:
Do you scale down the thumbnail or do we take whatever they supply?

Reply:
I'd prefer if we just passed in whatever the user sends without modifying the thumbnail in any way. This should also eliminate the need to package in any image manipulation libraries if `add_thumbnails` is turned off.

Reply:
I agree with that in principle except now we have an attack vector.

Reply:
For the attack vector situation I guess I would be OK with attempting to decode and check if it is a legitimate image (similar to imagemagick's verify command), however it would be non-optimal if we had to re-encode an already-lossy JPEG thumbnail and increase compression artifacts. Additionally, there was a bug where we didn't write the EXIF orientation tag in the thumbnail, so if decided to re-encode any supplied thumbnails we would need to make sure to copy all of the pertinent headers over.

Reply:
It was always the case that if you disable thumbnail generation, you have to set your own thumbnails. Nikon is using that feature. This just makes the feature work as intended if add_thumbnails is enabled. If we are not generating the thumbnails, then we should just take whatever they supply. Part of the point here is to not introduce the overhead of our default thumbnail generation code.
@dkozma FYI, disabling add_thumbnails will remove the image library and all the associated code. I agree that I'd like to get away from having default features, but this seems like a pretty common one to expect to work by default. I'm torn on that.

Reply:
@mauricefisher64 At this level of the SDK there's only so much we can do to avoid misuse. We would either need to force developers to only use our thumbnail generation, or trust them to do the right thing.
FYI, I also allow overriding the thumbnail generator via options in from_file_with_options, but I may remove that since it feels redundant.

Reply:
@gpeacock I would expect most implementors to already have the libraries and tools to generate thumbnails in their codebase. Furthermore, we can't expect add_thumbnails to work with the plethora of video codecs and other documents we are planning on supporting.

Reply:
Well what we will see is manifests that are not performant and present a bad experience.  Even if we don't scale we should check the size to make sure it is reasonable.

Reply:
@dkozma Not sure I understand the comment.  There are only a few thumbnail formats supported in the spec.  It is not open-ended.